File: library/src/main/java/zjm/cst/dhu/library/api/repository/Repository.java
package zjm.cst.dhu.library.api.repository;

import java.util.List;
import java.util.Map;

import okhttp3.ResponseBody;
import retrofit2.Retrofit;
import rx.Observable;
import zjm.cst.dhu.library.api.ApiService;
import zjm.cst.dhu.library.mvp.model.NewsData;
import zjm.cst.dhu.library.mvp.model.NewsInf;

/**
 * Created by zjm on 2017/5/10.
 */

public class Repository implements BaseRepository {
    private ApiService mApiService;

    public Repository(Retrofit retrofit) {
        mApiService = retrofit.create(ApiService.class);
    }

    @Override
    public Observable<Map<String, List<NewsData>>> getNewsList(String type, String id, int start) {
        return mApiService.getNewsList(type, id, start);
    }

    @Override
    public Observable<Map<String, NewsInf>> getNewInf(String postId) {
        return mApiService.getNewInf(postId);
    }

    @Override
    public Observable<ResponseBody> getNewsInfImg(String path) {
        return mApiService.getNewsInfImg(path);
    }
}

File: library/src/main/kotlin/zjm/cst/dhu/library/api/repository/Repository.kt
package zjm.cst.dhu.library.api.repository

import okhttp3.ResponseBody
import zjm.cst.dhu.library.api.ApiService
import zjm.cst.dhu.library.mvp.model.NewsData
import zjm.cst.dhu.library.mvp.model.NewsInf
import retrofit2.Retrofit
import rx.Observable

/**
 * Created by zjm on 2017/5/10.
 */

class Repository(retrofit: Retrofit)
    : BaseRepository {

    private val mApiService: ApiService = retrofit.create(ApiService::class.java)

    override fun getNewsList(type: String, id: String, start: Int):
            Observable<Map<String, List<NewsData>>> = mApiService.getNewsList(type, id, start)

    override fun getNewInf(postId: String):
            Observable<Map<String, NewsInf>> = mApiService.getNewInf(postId)

    override fun getNewsInfImg(path: String):
            Observable<ResponseBody> = mApiService.getNewsInfImg(path)

}
